core/src/main/java/com/oath/oak/NativeAllocator/OakNativeMemoryAllocator.java

```diff
@@ -115,7 +115,7 @@ public ByteBuffer allocate(int size) {
     public void free(ByteBuffer bb) {
         allocated.addAndGet(-(bb.remaining()));
         if (stats != null) stats.release(bb);
-        freeList.add(new Pair<>(System.identityHashCode(bb), bb));
+        freeList.add(new Pair<>(freeCounter.getAndIncrement(), bb));
```

Comment:
No, there was no bug because every released bb is unique.
I found this was a performance issue, the identityHashCode function is heavy and its faster to just use an atomic counter